src/token_position.ts

```diff
@@ -267,64 +268,109 @@ export function findElementWithOffset(
 
     const textNodes = getTextNodes(codeElement)
 
-    // How far forward we have looked so far. Starting at one because codeintellify treats positions as being 1-indexed.
-    let offsetStep = 1
-    let nodeIndex = 0
-
-    // Find the text node that is at the given offset.
-    let targetNode: Node | undefined
-    for (const [i, node] of textNodes.entries()) {
-        const text = node.textContent ?? ''
-        if (offsetStep <= offset && offsetStep + text.length > offset) {
-            targetNode = node
-            nodeIndex = i
-            break
+    let startNode: Node
+    let endNode: Node
+    if (typeof offsetEnd === 'number' && offsetEnd > offsetStart) {
```

Comment:
**question**: why do ranges and positions need to be handled in completely separate branches?

Reply:
There's not much overlap between these branches. When `findElementWithOffset` is called with a `Position`, we should look forwards and backwards to find all elements that comprise the token at that position. When it's called with a `Range` we assume that the position boundaries are intentional (i.e. that the user either already knows the start and end positions of the token, or doesn't want to highlight a discrete token)